pkg/controllers/nodeclass/controller.go

```diff
@@ -80,6 +95,31 @@ func (c *Controller) Reconcile(ctx context.Context, nodeClass *v1beta1.EC2NodeCl
 	return reconcile.Result{RequeueAfter: 5 * time.Minute}, err
 }
 
+func (c *Controller) Finalize(ctx context.Context, nodeClass *v1beta1.EC2NodeClass) (reconcile.Result, error) {
+	stored := nodeClass.DeepCopy()
+	if !controllerutil.ContainsFinalizer(nodeClass, v1beta1.TerminationFinalizer) {
+		return reconcile.Result{}, nil
+	}
+	nodeClaimList := &corev1beta1.NodeClaimList{}
+	if err := c.kubeClient.List(ctx, nodeClaimList, client.MatchingFields{"spec.nodeClass.name": nodeClass.Name}); err != nil {
+		return reconcile.Result{}, fmt.Errorf("listing NodeClaims using NodeClass, %w", err)
```

Comment:
nit on capitalization
```suggestion
		return reconcile.Result{}, fmt.Errorf("listing node claims using node class, %w", err)
```